apps/judicial-system/web/src/routes/Court/Indictments/CourtRecord/CourtRecord.tsx

```diff
@@ -47,14 +57,42 @@ const CourtRecord: React.FC = () => {
     caseNotFound,
   } = useContext(FormContext)
   const [navigateTo, setNavigateTo] = useState<keyof stepValidationsType>()
+  const [displayFiles, setDisplayFiles] = useState<TUploadFile[]>([])
+  const [allFilesUploaded, setAllFilesUploaded] = useState<boolean>(true)
 
   const { formatMessage } = useIntl()
   const { transitionCase } = useCase()
 
-  const { files, handleS3Upload, handleRetry, allFilesUploaded } = useS3Upload(
-    workingCase,
+  const { upload, remove } = useS3Upload(workingCase.id)
+
+  useEffect(() => {
+    if (workingCase.caseFiles) {
+      setDisplayFiles(workingCase.caseFiles.map(mapCaseFileToUploadFile))
+    }
+  }, [workingCase.caseFiles])
+
+  useMemo(() => {
+    setAllFilesUploaded(
+      displayFiles.every(
+        (file) => file.status === 'done' || file.status === 'error',
+      ),
+    )
+  }, [displayFiles, setAllFilesUploaded])
```

Comment:
Consider refactoring this to not use state.
```suggestion
  const allFilesUploaded = useMemo(() => 
      displayFiles.every(
        (file) => file.status === 'done' || file.status === 'error',
      ), [displayFiles])
```

Reply:
Awesome! Thanks 🙏